app/views/settings/index.html.erb

```diff
@@ -14,8 +14,15 @@
             <%= f.label :email %>
             <%= f.email_field :email, class: "form-control input-lg" %>
           </div>
+          <div class="form-group">
+            <%= f.label :uid, "Twitter UID" %>
+            <%= f.text_field :uid, class: "form-control input-lg" %>
+          </div>
           <div class="actions">
-            <%= f.submit "Update", class: "btn-lg btn-primary btn-block" %>
+            <%= f.submit "Update", class: "btn btn-lg btn-primary" %>
+            <% if !@settings.linked_with_twitter? %>
+            <%= link_to('Link Twitter Account', '/auth/twitter', class: 'btn btn-lg btn-primary pull-right') %>
```

Comment:
Calls User. linked_with_twitter? to determine whether to show the "Link Twitter Account" button.